metrics/metrics.go

```diff
@@ -45,32 +46,42 @@ func Middleware(registry prometheus.Registerer) gin.HandlerFunc {
 		Name:      "request_duration_seconds",
 		Help:      "A histogram of duration, in seconds, handling HTTP requests.",
 		Buckets:   prometheus.ExponentialBuckets(0.001, 2, 15),
-	}, []string{"host", "method", "path", "status"})
+	}, []string{"host", "method", "path", "status", "blocking"})
 
-	requestCount := prometheus.NewGauge(prometheus.GaugeOpts{
+	requestCount := prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: "http",
 		Name:      "requests_active",
 		Help:      "A gauge of the number of requests currently being handled.",
-	})
+	}, []string{"blocking"})
 
 	registry.MustRegister(requestDuration, requestCount)
 
 	return func(c *gin.Context) {
-		requestCount.Inc()
-		defer requestCount.Dec()
+		blocking := blockingRequestLabel(c.Request)
+
+		count := requestCount.With(prometheus.Labels{"blocking": blocking})
+		count.Inc()
+		defer count.Dec()
 
 		t := time.Now()
 		c.Next()
 
 		requestDuration.With(prometheus.Labels{
-			"host":   c.Request.Host,
-			"method": c.Request.Method,
-			"path":   c.FullPath(),
-			"status": strconv.Itoa(c.Writer.Status()),
+			"host":     c.Request.Host,
+			"method":   c.Request.Method,
+			"path":     c.FullPath(),
+			"status":   strconv.Itoa(c.Writer.Status()),
+			"blocking": blocking,
 		}).Observe(time.Since(t).Seconds())
 	}
 }
 
+func blockingRequestLabel(req *http.Request) string {
+	rawIndex := req.URL.Query().Get("lastUpdateIndex")
+	updateIndex, _ := strconv.ParseInt(rawIndex, 10, 64)
+	return strconv.FormatBool(updateIndex > 0)
```

Comment:
Instead of parsing this as an int, I think I'm going to simplify this:

```suggestion
	return strconv.FormatBool(rawIndex != "0")
```

Any non-zero value should be fine to consider this a blocking request.

Reply:
hmm, although I still need to consider the `""` case as well